README.md

```diff
@@ -124,15 +126,79 @@ flyway -user=$USER -password="" -locations="filesystem:database/sql/" -url=jdbc:
+            hardhat: {
+                initialBaseFeePerGas: 0,
+                accounts: {
+                    accountsBalance: "1000000000000000000000000"
+                }
+            }
+        }
+};</code></pre>
+4. Run local testnet with `npx hardhat node`
+
+## Running the Services Locally
+
+### Prerequisites
+Reading the state of the blockchain requires issuing RPC calls to an ethereum node. This can be a testnet you are running locally, some "real" node you have access to or the most convenient thing is to use a third party service like [infura](https://infura.io/) to get access to an ethereum node which we recommend.
+After you made a free infura account they offer you "endpoints" for the mainnet and different testnets. We will refer those as `node-urls`.
+Because Gnosis only runs their services on mainnet, rinkeby and gnosis chain you need to select one of those.
+
+Note that the `node-url` is sensitive data. The `orderbook` and `solver` executables allow you to pass it with the `--node-url` parameter. This is very convenient for our examples but to minimize the possibility of sharing this information by accident you should consider setting the `NODE_URL` environment variable so you don't have to pass the `--node-url` argument to the executables.
+
+To avoid confusion during your tests, always double check that the token and account addresses you use actually correspond to the network of the `node-url` you are running the executables with.
+
+### Orderbook
+
+To see all supported command line arguments run `cargo run --bin orderbook -- --help`.
+
+Run an `orderbook` on `localhost:8080` with:
+
+```sh
+cargo run --bin orderbook -- \
+  --skip-trace-api true \
+  --node-url <YOUR_NODE_URL>
+```
+
+`--skip-trace-api true` will make the orderbook compatible with more ethereum nodes. If your node supports `trace_callMany` you can drop this argument.
+
+### Solvers
+
+To see all supported command line arguments run `cargo run --bin solver -- --help`.
+
+Run a solver which is connected to an `orderbook` at `localhost:8080` with:
 
 ```sh
-cd contracts; cargo run --bin deploy --features bin; cd -
+cargo run -p solver -- \
+  --solver-account 0xa6DDBD0dE6B310819b49f680F65871beE85f517e \
+  --baseline-sources UniswapV2,SushiSwap \
+  --node-url <YOUR_NODE_URL>
 ```
 
-## Running the services
+Because this command was designed to work with `node-urls` from mainnet and rinkeby it limits the `baseline-sources` to `UniswapV2` and `SushiSwap`. If your `node-url` belongs to mainnet, you can drop this argument all together to get access to all supported `baseline-sources`.
+
+The `solver-account` is responsible for signing transactions. Solutions for settlements need to come from an address the settlement contract trusts in order to make the contract actually consider the solution. Adding your personal solver account is quite involved and requires you to get in touch with the team, so we are using this public solver address for now.
```

Comment:
Maybe add a note that we are:
> pretending to use the public solver account

Since you can't actually use it without a private key.